packages/run-protocol/test/vaultFactory/test-liquidator.js

```diff
@@ -342,19 +342,73 @@ const makeDriver = async (t, initialPrice, priceBase) => {
     E(lender).getCollateralManager(aethBrand),
   ).getNotifier();
   let managerNotification = await E(managerNotifier).getUpdateSince();
-  /** @type {UserSeat<VaultKit>} */
-  let vaultSeat;
-  /** @type {VaultKit} */
-  let vault;
+
   /** @type {UserSeat<VaultKit>} */
   let lastSeat;
-  let notifier;
   let notification = {};
   let lastOfferResult;
+  const makeVaultDriver = async (collateral, debt) => {
+    /** @type {UserSeat<VaultKit>} */
+    const vaultSeat = await E(zoe).offer(
+      await E(lender).makeVaultInvitation(),
+      harden({
+        give: { Collateral: collateral },
+        want: { RUN: debt },
+      }),
+      harden({
+        Collateral: aethMint.mintPayment(collateral),
+      }),
+    );
+    const {
+      vault,
+      publicNotifiers: { vault: notifier },
+    } = await E(vaultSeat).getOfferResult();
+    t.truthy(await E(vaultSeat).hasExited());
+    return {
+      vault: () => vault,
+      vaultSeat: () => vaultSeat,
+      notification: () => notification,
+      close: async () => {
+        lastSeat = await E(zoe).offer(E(vault).makeCloseInvitation());
+        lastOfferResult = await E(lastSeat).getOfferResult();
+        t.is(
+          lastOfferResult,
+          'your loan is closed, thank you for your business',
+        );
+        t.truthy(await E(vaultSeat).hasExited());
+      },
+      checkNotify: async (phase, expected) => {
+        notification = await E(notifier).getUpdateSince();
+        trace(t, 'notify', notification);
+        t.is(notification.value.vaultState, phase);
+        expected && t.like(notification.value, expected);
```

Comment:
this expression style is common within larger declarations but I've rarely seen it in imperative. I thought this is preferred here,
```
if (expected) {
  t.like(notification.value, expected);
}
```
Is this an exception for test code or am I just wrong? :)  if so, fyi @samsiegart who I asked recently to remove a use of `&&`.

Either way, I'd argue that this would be better so as to allow testing for falsy expectations,
```
// note that notifiers can send `undefined` as the actual state but this driver doesn't handle that case
if (expected !== undefined) {
  t.like(notification.value, expected);
}
```

Reply:
In all these cases, the notifications are records. So I'm going to keep it focused on "the details of the record". They can always look at the value explicitly, which seems good enough for now.